app/screens/send-bitcoin-screen/send-bitcoin-destination-screen.tsx

```diff
@@ -240,14 +271,37 @@ const SendBitcoinDestinationScreen: React.FC<Props> = ({ route }) => {
     }
   }, [destinationState, goToNextScreenWhenValid, navigation, setGoToNextScreenWhenValid])
 
-  const initiateGoToNextScreen = useMemo(() => {
-    return async () => {
-      if (validateDestination) {
-        await validateDestination(destinationState.unparsedDestination)
-        setGoToNextScreenWhenValid(true)
+  // setTimeout here allows for the main JS thread to update the UI before the long validateDestination call
+  const waitAndValidateDestination = React.useCallback(
+    (input: string) => {
+      function returnValidateDestinationFunc(input: string, obj: NetworkWalletsContacts) {
+        return async function _validateDestination() {
+          await validateDestination(input, obj)
+        }
       }
+
+      setTimeout(
+        returnValidateDestinationFunc(input, {
+          _bitcoinNetwork: bitcoinNetwork,
+          _wallets: wallets,
+          _contacts: contacts,
+        }),
+        0,
+      )
+    },
+    [bitcoinNetwork, wallets, contacts, validateDestination],
+  )
```

Comment:
```suggestion
  const waitAndValidateDestination = React.useCallback(
    (input: string) => {
      setTimeout(
        () =>
          validateDestination(input, {
            _bitcoinNetwork: bitcoinNetwork,
            _wallets: wallets,
            _contacts: contacts,
          }),
        0,
      )
    },
    [bitcoinNetwork, wallets, contacts, validateDestination],
  )
```

I dont think `returnValidateDestinationFunc` is really necessary

Reply:
ehhhh maybe. I was having difficulty with variable input being stale when sent through. I'll see if I can remove that part, it could've been overkill.

Reply:
yea was overkill. removed it.